src/routes/auth/signDataItem.tsx

```diff
@@ -0,0 +1,189 @@
+import { useStorage } from "@plasmohq/storage/hook";
+import { ExtensionStorage } from "~utils/storage";
+import { checkPassword } from "~wallets/auth";
+
+interface Tag {
+  name: string;
+  value: string;
+}
+
+interface DataStructure {
+  data: number[];
+  tags: Tag[];
+}
+
+export default function SignDataItem() {
+  // connect params
+  const params = useAuthParams<{
+    appData: { appURL: string };
+    data: DataStructure;
+  }>();
+
+  const [password, setPassword] = useState<boolean>(false);
+  const { setToast } = useToasts();
+
+  const recipient =
+    params?.data?.tags.find((tag) => tag.name === "Recipient")?.value || "NA";
+  const quantity =
+    params?.data?.tags.find((tag) => tag.name === "Quantity")?.value || "NA";
+
+  const [signatureAllowance] = useStorage(
+    {
+      key: "signatureAllowance",
+      instance: ExtensionStorage
+    },
+    10
+  );
+
+  // get auth utils
+  const { closeWindow, cancel } = useAuthUtils("signDataItem", params?.authID);
+
+  const passwordInput = useInput();
+
+  const svgieAvatar = useMemo(
+    () => svgie(recipient, { asDataURI: true }),
+    [recipient]
+  );
+
+  useEffect(() => {
+    if (signatureAllowance < Number(quantity)) setPassword(true);
+  }, [signatureAllowance, quantity]);
+
+  // listen for enter to reset
+  useEffect(() => {
+    const listener = async (e: KeyboardEvent) => {
+      if (e.key !== "Enter") return;
+      await sign();
+    };
+
+    window.addEventListener("keydown", listener);
+
+    return () => window.removeEventListener("keydown", listener);
+  }, [params?.authID]);
+
+  // sign message
+  async function sign() {
+    if (password) {
+      const checkPw = await checkPassword(passwordInput.state);
+      if (!checkPw) {
+        setToast({
+          type: "error",
+          content: browser.i18n.getMessage("invalidPassword"),
+          duration: 2400
+        });
+        return;
+      }
+    }
+
+    // send response
+    await replyToAuthRequest("signDataItem", params?.authID);
+
+    // close the window
+    closeWindow();
+  }
+
+  return (
+    <Wrapper>
+      <div>
+        <HeadV2
+          title={browser.i18n.getMessage("sign_item")}
+          showOptions={false}
+          back={cancel}
+        />
+        <Description>
+          <Text noMargin>
+            {browser.i18n.getMessage(
+              "sign_data_description",
+              params?.appData.appURL
+            )}
+          </Text>
+          <div>
+            <ListItem
+              small
+              disabled
+              description={formatAddress(recipient, 10)}
+              title={"Recipient"}
+              img={svgieAvatar}
+              onClick={() =>
+                browser.tabs.create({
+                  url: `https://viewblock.io/arweave/address/${recipient}`
+                })
+              }
+            />
+            {/* TODO: will we have access to process ID? if so would be great to get the token image and redirect to ao scanner */}
```

Comment:
We should have the `Target` and `Action` tag within that same message/DataItem to fetch the Token's Info